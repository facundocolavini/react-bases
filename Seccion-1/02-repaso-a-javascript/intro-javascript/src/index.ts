'estoy en async await seccion 3'
export {}