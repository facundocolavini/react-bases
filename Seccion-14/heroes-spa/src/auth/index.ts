export * from './pages';
export * from './routes/AuthRoutes';
export * from './context';