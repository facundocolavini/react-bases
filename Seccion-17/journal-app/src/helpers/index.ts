export * from './fileUpload';
export * from './getActiveNote';
export * from './getEnviroments';
export * from './loadNotes';
export * from './promiseAll';

