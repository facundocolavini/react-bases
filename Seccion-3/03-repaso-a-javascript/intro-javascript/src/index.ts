
console.log('Fin de la seccion 3');

export {}