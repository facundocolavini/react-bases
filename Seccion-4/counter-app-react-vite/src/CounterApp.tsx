import PropTypes from 'prop-types'
import { useState } from 'react';

type ValuesProps  = {
   value: number;

}

export const CounterApp = ({value}: ValuesProps): JSX.Element => {
    
    const [counter , setCounter] = useState(value)

    const handleAdd =(): void => {
        // 
        setCounter(counter + 1)
        //Otra forma :
        // setCounter((c)=>  c + 1) // valor actual del counter + 1
    }

    const handleSubstract = (counter: number): void => {
        setCounter(counter - 1)
    }

    const handleReset = (): void=>{
        setCounter(value)
    }
    
    return (
        <>
            <h1>CounterApp</h1>
            <h2 data-testid="counter">{counter}</h2>
            <button aria-label="btn-increment" onClick={handleAdd}>+1</button>
            <button aria-label="btn-decrement" onClick={()=>handleSubstract(counter)}>-1</button>
            <button aria-label="btn-reset" onClick={handleReset}>Reset</button>
        </>
    )
}

CounterApp.propTypes ={
    value: PropTypes.number.isRequired,
}