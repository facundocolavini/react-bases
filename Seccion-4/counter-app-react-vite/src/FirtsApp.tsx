//Proptypes 
import PropTypes from 'prop-types'


// Variables

const newMessage : string ='Facundo';// No se rerenderiza
const newArray : Array<number> = [1,2,3,4,5,6,7,8,9];// No se rerenderiza
interface Message {
    message: string;
    title: string;
}
const msgObj: Message = {
    message: 'Hola Mundo',
    title: 'Facundo'
}
//Recomendado si no tiene una dependencia del componente dejar las funciones por fuera 
function sayHi(message:string): string{
        return message 
}


// Doble validacion si es que hay propTypes por eso hago las props opcionales ?
interface Titles {
    title?: string;
    subTitle?: string; 
    name?: string;
}

export const FirtsApp: React.FC<Titles> = ({title,subTitle, name}) => {
/*     if(!title) {
        throw new Error('El titulo no existe');
    } */
    
    return (
    <>
        <h1>{title}</h1>
        {/* <code>{ JSON.stringify(msgObj) }</code> */}
        <p>{subTitle}</p>
        <p>{name}</p>
    </>
  )
}

// PropTypes
FirtsApp.propTypes = {
    title: PropTypes.string.isRequired,
    subTitle: PropTypes.string.isRequired, // Si uso interfaces y propTypes estaria manejando 2 veces la validacion de props
}

//Default Propts entran antes de nuestro propTypes
FirtsApp.defaultProps = { // Si utilizo interface al poner valores default que no existen en mi interface dara un error
    title: 'No hay titulo',
    subTitle: 'No hay subtitulo',
    name: 'Facundo Colavini'
}

