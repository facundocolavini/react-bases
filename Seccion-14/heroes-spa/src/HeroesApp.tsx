import { AuthProvider } from "./auth"
import { AppRouter } from "./router/AppRouter"

interface AppHeroes {

}

export const HeroesApp = (props: AppHeroes) => {
  return (
    <AuthProvider>
      <AppRouter />
    </AuthProvider>
  )
}