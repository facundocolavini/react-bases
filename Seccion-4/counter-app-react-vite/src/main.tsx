import React from 'react';
import ReactDOM from 'react-dom/client'
import { CounterApp } from './CounterApp';
import { FirtsApp } from './FirtsApp';
import { HelloWorldApp } from './HelloWorldApp';
//Styles 
import './styles.css'




ReactDOM.createRoot(document.getElementById('root') as HTMLElement).render(
    <React.StrictMode>
        {/* <FirtsApp></FirtsApp> */}
        <CounterApp value={1} />
    </React.StrictMode>
  )