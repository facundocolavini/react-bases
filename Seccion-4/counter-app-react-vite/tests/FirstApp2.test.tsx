
import * as React from 'react'
import { describe, expect, test } from '@jest/globals';
import { render, screen } from '@testing-library/react';
import { FirtsApp } from '../src/FirstApp';

describe('Prueba en <FirstApp/>',()=>{
    //Podemos declarar las variables o constantes aca para no repetir codigo 
    const title: string  =  'Hola , Soy Goku';
    const subTitle: string = 'Soy un subtitulo';

    test('Debe hacer match con el snapshot',()=>{
        const { container } =  render(<FirtsApp title={title}/>)
        expect(container).toMatchSnapshot()
    })

    test('Debe de mostrar el mensaje "Hola , Soy Goku"', () => { 
        // No hace falta el container para esta prueba
        render(<FirtsApp title={title}/>)        

        expect(screen.getByText(title)).toBeTruthy()

    })

    test("Debe mostrar el titulo en un H1" ,()=>{
        render(<FirtsApp title={title}/>)
        expect(screen.getByRole('heading',{level:1}).innerHTML).toContain(title) // Va buscar el primer H1 de nivel 1 de mi html.
    })

    test('Debe mostrar el subtitulo enviado por props',()=>{
        render(<FirtsApp title={title} subTitle={subTitle}/>)
        expect( screen.getAllByText(subTitle).length ).toBe(2)
    })
})