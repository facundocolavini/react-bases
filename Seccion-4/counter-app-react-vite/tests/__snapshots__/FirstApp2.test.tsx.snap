// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba en <FirstApp/> Debe hacer match con el snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
     
    Hola , Soy Goku
     
  </h1>
  <p>
    No hay subtitulo
  </p>
  <p>
    No hay subtitulo
  </p>
  <p>
    Facundo Colavini
  </p>
</div>
`;
